### PR TITLE
File form fields update

### PR DESCRIPTION
**Pull request by devkabiir**
This will allow users to have `File` from fields, example:

```dart
  /// Do not forget to add the appropriate headers
  @Headers({'Content-Type': 'multipart/form-data'})
  @POST("/profile")

  /// This will add the image name from `image.path.split(Platform.pathSeperator).last`
  Future<Response<String>> setProfileImage(@Field() File image);
```

### CODE REVIEW

File: annotation/README.md [hunk 1]
```diff
@@ -66,6 +66,24 @@ abstract class RestClient {
       {@QueryMap() Map<String, dynamic> queryies,
       @Field() int field,
       @Field("field-g") String ffff});
+
+  /// Do not forget to add the appropriate headers
+  @Headers({'Content-Type': 'multipart/form-data'})
```
Comment (trevorwang) [inline annotation/README.md]: How about using a `Multipart` annotation instead?

Comment (trevorwang) [inline annotation/README.md]: How about using a `Multipart` annotation instead?

Comment (devkabiir) [reply annotation/README.md]: We can add another annotation sure. I will update the PR

Comment (devkabiir) [reply annotation/README.md]: Actually dio automatically adds `Content-Type` header when we send form data. Look at this -> https://github.com/flutterchina/dio/blob/master/package_src/lib/src/dio.dart#L848-L851

So I will remove the `@Headers` from the example

Comment (trevorwang) [reply annotation/README.md]: make sense.

**Pull request merged**